### PR TITLE
[SIG 4416] Show nearby incidents on object map

### PR DESCRIPTION
**Pull request by kiruys**
In asset select maps (for example a container select map), it is now possible to see if an asset already has incidents.
When an asset on the map is clicked, the oldest incident related to that asset becomes visible in the detail panel.
Incidents not related to assets (i.e. with a different location) are shown with markers.

Previously, incidents related to assets were visible by incident markers placed on top of the asset. To avoid this, incident markers are not rendered when the coordinates are identical to the coordinates of one of the assets.

## Signalen

Before opening a pull request, please ensure:

- [x] Double-check your branch is based on `develop` and targets `develop`
- [x] Pull request has tests (we are going for 100% coverage!)
- [x] Code is well-commented, linted and follows project conventions
- [x] Committed source code is headed by the correct SPDX license expression

Be kind to code reviewers, please try to keep pull requests as small and focused as possible :)

### CODE REVIEW

File: src/signals/incident/components/form/MapSelectors/Asset/Selector/DetailPanel/DetailPanel.tsx [hunk 1]
```diff
@@ -54,17 +60,39 @@ const nearbyLegendItem = {
...
+}
+
+type Properties = {
+  category: {
+    name: string
+  }
+  created_at: string
+}
+
+interface SelectionIncident {
```
Comment (OscarBakker) [inline src/signals/incident/components/form/MapSelectors/Asset/Selector/DetailPanel/DetailPanel.tsx:75]: Why is Properties a type and interface a SelectionIncident?

File: src/signals/incident/components/form/MapSelectors/Asset/Selector/DetailPanel/DetailPanel.tsx [hunk 2]
```diff
@@ -158,6 +186,39 @@ const DetailPanel: FC<DetailPanelProps> = ({ language = {} }) => {
...
+    setSelectionIncident({})
+
+    if (selectionNearby) {
+      setSelectionIncident({
+        categoryName: selection?.label,
+        createdAt: selection?.description,
+      })
+    }
+
+    if (selectionOnMap && data?.features) {
```
Comment (OscarBakker) [inline src/signals/incident/components/form/MapSelectors/Asset/Selector/DetailPanel/DetailPanel.tsx:214]: FeatureCollection does seem to make features required, so optional chain could be removed?!

File: src/signals/incident/components/form/MapSelectors/Asset/Selector/NearbyLayer/NearbyLayer.tsx [hunk 3]
```diff
@@ -148,6 +161,9 @@ export const NearbyLayer: FC<NearbyLayerProps> = ({ zoomLevel }) => {
 
     data.features.forEach((feature) => {
       const { lat, lng } = featureToCoordinates(feature.geometry)
+
+      if (findAssetMatch(assetData, lat, lng)) return
```
Comment (OscarBakker) [inline src/signals/incident/components/form/MapSelectors/Asset/Selector/NearbyLayer/NearbyLayer.tsx:165]: maybe a small description

File: src/signals/incident/components/form/MapSelectors/Asset/Selector/DetailPanel/DetailPanel.tsx (hunk 1 above)
Comment (kiruys) [reply src/signals/incident/components/form/MapSelectors/Asset/Selector/DetailPanel/DetailPanel.tsx:75]: I will change SelectionIncident to a type to be more consistent

File: src/signals/incident/components/form/MapSelectors/Asset/Selector/DetailPanel/DetailPanel.tsx [hunk 4]
```diff
@@ -54,17 +60,39 @@ const nearbyLegendItem = {
...
+}
+
 const DetailPanel: FC<DetailPanelProps> = ({ language = {} }) => {
   const shouldRenderAddressPanel = useMediaQuery({
     query: breakpoint('max-width', 'tabletM')({ theme: ascDefaultTheme }),
   })
   const [showLegendPanel, setShowLegendPanel] = useState(false)
   const [optionsList, setOptionsList] = useState(null)
+  const [selectionIncident, setSelectionIncident] = useState<SelectionIncident>(
+    {}
```
[the remark this answers is not in the document]
Comment (kiruys) [reply src/signals/incident/components/form/MapSelectors/Asset/Selector/DetailPanel/DetailPanel.tsx:87]: certainly not

[the remark this answers is not in the document]
Comment (kiruys) [reply src/signals/incident/components/form/MapSelectors/Asset/Selector/DetailPanel/DetailPanel.tsx:87]: I've tried to remove it, but it returns automatically🥲

File: src/signals/incident/components/form/MapSelectors/Asset/Selector/DetailPanel/DetailPanel.tsx (hunk 2 above)
Comment (kiruys) [reply src/signals/incident/components/form/MapSelectors/Asset/Selector/DetailPanel/DetailPanel.tsx]: If I remove it, the build fails

File: src/signals/incident/components/form/MapSelectors/Asset/Selector/NearbyLayer/NearbyLayer.tsx (hunk 3 above)
Comment (kiruys) [reply src/signals/incident/components/form/MapSelectors/Asset/Selector/NearbyLayer/NearbyLayer.tsx:165]: I've added a description and `return null`

**Review by OscarBakker — APPROVED** [review]

**Pull request merged**